cms/django_helpers.py

```diff
@@ -0,0 +1,76 @@
+# Core packages
+import os
+
+# Third party packages
+import yaml
+from django.shortcuts import redirect, render
+from django.urls import re_path
+from yamlloader import ordereddict
+
+
+def _create_view(view_callback, url_mapping, settings={}):
+    """
+    Create a view function to execute the callback function with the
+    url_arguments
+    """
+
+    def url_view(request, *args, **kwargs):
+        return view_callback(request, url_mapping, settings, *args, **kwargs)
+
+    return url_view
+
+
+def _create_views_from_yaml(yaml_filepath, view_callback, settings={}):
+    """
+    Givan a YAML file mapping URL paths to values, e.g.:
+
+        path/one: {"some": "value"}
+        path/two: {"another": "value"}
+
+    Create a Django URL pattern from each value, so that when that path
+    is requested, view_callback is run, passing the mapped value.
+
+    Returns a list of Django urlpatterns.
+    """
+
+    urlpatterns = []
+
+    if os.path.isfile(yaml_filepath):
+        with open(yaml_filepath) as yaml_paths_file:
+            url_paths = yaml.load(
+                yaml_paths_file.read(), Loader=ordereddict.CLoader
+            )
+            if url_paths:
+                for url_path, url_mapping in url_paths.items():
+                    urlpatterns.append(
+                        re_path(
+                            r"^{0}$".format(url_path),
+                            _create_view(view_callback, url_mapping, settings),
+                        )
+                    )
+
+    return urlpatterns
+
+
+def _redirect_to_target(request, url_mapping, settings, *args, **kwargs):
+    location = url_mapping.format(**kwargs)
+    query = request.META["QUERY_STRING"]
+
+    if query:
+        location += "?" + query
+
+    return redirect(location, permanent=settings.get("permanent", False))
+
+
+def _deleted_callback(request, url_mapping, settings, *args, **kwargs):
+    return render(request, "410.html", url_mapping, status=410)
+
+
+def create_redirect_views(path="redirects.yaml", permanent=False):
```

Comment:
Isn't this provided by [through `flask-base`](https://github.com/canonical/canonicalwebteam.flask-base/blob/main/canonicalwebteam/flask_base/app.py#L217) by [yaml-responses](https://github.com/canonical/canonicalwebteam.yaml-responses)? Do we need to include it here?

Reply:
But from the diff it looks like you added this in. Isn't that right? So what led you to add it & where did you get it from?

Reply:
And yes, try taking it out and see if `redirects.yaml`and `deleted.yaml` still work.

Reply:
Alright

Reply:
I think the diff was due to the linting though

Reply:
Looks like the entire file exists in canonicalwebteam.yaml-responses so we can take it out

Reply:
Or maybe not.. The latest release of canonicalwebteam.yaml-responses library is using an older version of [url imports](https://github.com/canonical/canonicalwebteam.yaml-responses/blob/c69fbcc081a676806be3c44530b1111e84fe5e84/canonicalwebteam/yaml_responses/django_helpers.py#L7) which was removed in Django 4.0.
Could we leave the file in for now and then create another task to update the library and circle back to this afterwards?

Reply:
Yes let's leave it. Good work. :+1: